recipes/cassandra-cpp-driver/all/conanfile.py

```diff
@@ -0,0 +1,137 @@
+import os.path
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class CassandraCppDriverConan(ConanFile):
+    name = "cassandra-cpp-driver"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://docs.datastax.com/en/developer/cpp-driver/"
+    description = "DataStax C/C++ Driver for Apache Cassandra and DataStax Products"
+    topics = ("cassandra", "cpp-driver", "database", "conan-recipe")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "install_header_in_subdir": [True, False],
+        "use_atomic": [None, "boost", "std"],
+        "with_openssl": [True, False],
+        "with_zlib": [True, False],
+        "with_kerberos": [True, False],
+        "use_timerfd": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "install_header_in_subdir": False,
+        "use_atomic": None,
+        "with_openssl": True,
+        "with_zlib": True,
+        "with_kerberos": False,
+        "use_timerfd": True
+    }
+
+    generators = "cmake"
+    exports_sources = [
+        "CMakeLists.txt",
+        "patches/*"
+    ]
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["VERSION"] = self.version
+        cmake.definitions["CASS_BUILD_EXAMPLES"] = False
+        cmake.definitions["CASS_BUILD_INTEGRATION_TESTS"] = False
+        cmake.definitions["CASS_BUILD_SHARED"] = self.options.shared
+        cmake.definitions["CASS_BUILD_STATIC"] = not self.options.shared
+        cmake.definitions["CASS_BUILD_TESTS"] = False
+        cmake.definitions["CASS_BUILD_UNIT_TESTS"] = False
+        cmake.definitions["CASS_DEBUG_CUSTOM_ALLOC"] = False
+        cmake.definitions["CASS_INSTALL_HEADER_IN_SUBDIR"] = self.options.install_header_in_subdir
+        cmake.definitions["CASS_INSTALL_PKG_CONFIG"] = False
+
+        if self.options.use_atomic == "boost":
+            # Compilation error on Linux
+            if self.settings.os == "Linux":
+                raise ConanInvalidConfiguration(
+                    "Boost.Atomic is not supported on Linux at the moment")
```

Comment:
This should be in the configure method

Reply:
[Fixed](https://github.com/conan-io/conan-center-index/pull/3110/commits/a5aa11ac96c8401d120c0f9c747664e271231de0)